### PR TITLE
fix: coerce executeJavaScript arguments

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Hopefully this will result in less confusing failures when the wrong type of
thing is passed. Ref #24393.

Also fixes an issue with an incorrect argument list in `executeJavaScriptInIsolatedWorld` resulting in the 3rd argument (`hasUserGesture`) being ignored.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Fixed executeJavaScriptInIsolatedWorld not respecting `hasUserGesture` argument.

### CODE REVIEW

**Comment by codebytere** [comment]
Is this testable somehow?

File: lib/browser/api/web-contents.js [hunk 1]
```diff
@@ -207,11 +207,11 @@ const waitTillCanExecuteJavaScript = async (webContents) => {
...
 WebContents.prototype.executeJavaScript = async function (code, hasUserGesture) {
   await waitTillCanExecuteJavaScript(this);
-  return ipcMainUtils.invokeInWebContents(this, false, 'ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', 'executeJavaScript', code, hasUserGesture);
+  return ipcMainUtils.invokeInWebContents(this, false, 'ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', 'executeJavaScript', String(code), !!hasUserGesture);
 };
-WebContents.prototype.executeJavaScriptInIsolatedWorld = async function (code, hasUserGesture) {
+WebContents.prototype.executeJavaScriptInIsolatedWorld = async function (worldId, code, hasUserGesture) {
   await waitTillCanExecuteJavaScript(this);
-  return ipcMainUtils.invokeInWebContents(this, false, 'ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', 'executeJavaScriptInIsolatedWorld', code, hasUserGesture);
+  return ipcMainUtils.invokeInWebContents(this, false, 'ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', 'executeJavaScriptInIsolatedWorld', worldId, code, !!hasUserGesture);
```
Comment (miniak) [inline lib/browser/api/web-contents.js:214]: `String(code)`

Comment (nornagon) [reply lib/browser/api/web-contents.js:214]:
```suggestion
  return ipcMainUtils.invokeInWebContents(this, false, 'ELECTRON_INTERNAL_RENDERER_WEB_FRAME_METHOD', 'executeJavaScriptInIsolatedWorld', worldId, String(code), !!hasUserGesture);
```

Comment (nornagon) [reply lib/browser/api/web-contents.js:214]: ah now i remember why i took this out. `code` isn't a string for this function.

**Comment by nornagon** [comment]
> Is this testable somehow?

yeah, we can test that calling the function with weird argument types results in a more clear error than "Failed to serialize arguments"

**Comment by nornagon** [comment]
Ah, we should also probably test that `hasUserGesture` works in both forms of `executeJavaScript`.

**Pull request merged**